thinc/backends/cupy_ops.py

```diff
@@ -93,6 +93,8 @@ def asarray(self, data, dtype=None):
             array = tensorflow2xp(data)
         elif is_mxnet_gpu_array(data):
             array = mxnet2xp(data)
+        elif dtype is not None:
+            array = self.xp.array(data, dtype=dtype)
```

Comment:
Do we need this as a separate case? numpy/cupy.array have `None` as the default value of `dtype`. So, we could use the `else` branch and pass `dtype` there.

Reply:
Maybe not? I was mainly trying to revert it to something closer to the version before #656.